File: Task/Largest-proper-divisor-of-n/Fortran/largest-proper-divisor-of-n.f
       program LargestProperDivisors
       implicit none
       integer i, lpd
       do 10 i=1, 100
           write (*,'(I3)',advance='no') lpd(i)
 10        if (i/10*10 .eq. i) write (*,*)
       end program

       integer function lpd(n)
       implicit none
       integer n, i
       if (n .le. 1) then
           lpd = 1
       else
           do 10 i=n-1, 1, -1
 10            if (n/i*i .eq. n) goto 20
 20        lpd = i
       end if
       end function

File: Task/Largest-proper-divisor-of-n/PascalABC.NET/largest-proper-divisor-of-n.pas
##
function lpd(n: integer) := range((n / 2).Ceil, 1, -1).First(x -> n mod x = 0);

for var n := 1 to 100 do
  write(lpd(n):3, if n mod 10 = 0 then #10 else '');
